tokio/src/net/tcp/stream.rs

```diff
@@ -1096,10 +1104,17 @@ impl TcpStream {
     /// # Ok(())
     /// # }
     /// ```
+    #[cfg(not(target_os = "wasi"))]
     pub fn linger(&self) -> io::Result<Option<Duration>> {
         socket2::SockRef::from(self).linger()
     }
 
+    /// Returns `io::ErrorKind::Unsupported` as this is not supported on WebAssembly
+    #[cfg(target_os = "wasi")]
+    pub fn linger(&self) -> io::Result<Option<Duration>> {
+        Err(io::ErrorKind::Unsupported.into())
```

Comment:
Why have this method at all if it is not supported. The method could just be omitted on platforms that do not support it?

Reply:
Removed